app/controllers/concerns/paginable.rb

```diff
@@ -11,12 +11,75 @@ module Paginable
     def paginable_renderise(partial: nil, controller: params[:controller], action: params[:action], scope: nil, locals: {})
       raise ArgumentError, 'scope should be an instance of ActiveRecord::Relation class' unless scope.is_a?(ActiveRecord::Relation)
       raise ArgumentError, 'locals should be an instance of Hash' unless locals.is_a?(Hash)
+      @paginable_controller = controller
+      @paginable_action = action
+      refined_scope = refine_query(scope)
       render(layout: '/layouts/paginable', partial: partial, locals: { 
         controller: controller,
         action: action,
-        # The scope is paginable if it has been chained with page method from kaminari which contains methods such as total_pages
-        paginable: scope.respond_to?(:total_pages), 
-        scope: scope }.merge(locals))
+        paginable: refined_scope.respond_to?(:total_pages), 
+        scope: refined_scope }.merge(locals))
+    end
+    # Generates an URL to sort given a sort field.
+    # sort_field {String} - Represents the column name for a table
+    def paginable_sort_url(sort_field)
+      return link_to(sort_link_name(sort_field), sort_link_url(sort_field), 'data-remote': true)
     end
   end
+  private
+    # Returns the upcase string (e.g ASC or DESC) if sort_direction param is present in any of the forms 'asc', 'desc', 'ASC', 'DESC'
+    # otherwise returns nil
+    def sort_direction
+      if params.present? && params[:sort_direction].present?
+        directions = ['asc', 'desc', 'ASC', 'DESC']
+        return directions.include?(params[:sort_direction]) ? params[:sort_direction].upcase : 'ASC'
+      end
+      return nil
+    end
+    # Returns DESC when ASC is passed and vice versa, otherwise nil
+    def swap_sort_direction(direction)
+      return 'DESC' if direction == 'ASC'
+      return 'ASC' if direction == 'DESC'
+    end
+    # Refine a scope passed to this concern if any of the params (search, sort_field or page) are present
+    def refine_query(scope)
```

Comment:
Nice refactor. I'll update the template pagination PR to make sure template.rb model has a seach scope